Fix failing Calamari build

scottmerchant opened the pull request:
The build is failing because one call seems to return json as separate log lines instead of a single log line containing the json blob. Unsure why it has randomly started doing this as I can't see a reason.. but it could be the AWS CLI? Unsure if we updated a version of something that would affect this.

[sc-42546]

scottmerchant commented on source/Calamari.Tests/KubernetesFixtures/KubernetesRunsScriptFixture.cs:
```diff
@@ -238,7 +238,6 @@ public void ExecutionWithEKS_IAMAuthenticator()
         }
         
         [Test]
-        [WindowsTest] // This test requires the aws cli tools. Currently only configured to install on Windows
```
This is not true anymore, I don't think. I've removed it to see if the test passes on linux.

scottmerchant replied:
I was wrong - linux needs sudo to install the tool, which blows up the test on the workers. 😞

johnsimons replied:
This is interesting because we are using `sudo` in https://github.com/OctopusDeploy/Calamari/blob/d203d663372dd02ae33920ab420eabf2e3a5e99a/source/Calamari.Tests/KubernetesFixtures/InstallTools.cs#L166
Couldn't we do the same here too @scottmerchant ?

scottmerchant replied:
Ah yep I guess so - I didn't look to much into it as originally it wasn't running on linux so I just thought I'd try and see how it goes. I think it's probably worth just getting this change in to get the build green so Yihao can get his PR going. I'll fix the cli version as you suggested though 👍

scottmerchant commented:
I've added some logging because it looks like the output is not simply the JSON like it is on my local machine. Once I know what we receive I can add some code to parse it correctly.

scottmerchant commented on source/Calamari.Tests/KubernetesFixtures/Approved/KubernetesContextScriptWrapperFixture.ExecutionWithEKS_IAMAuthenticator.approved.txt:
```diff
@@ -8,7 +8,7 @@
 [Info] Creating kubectl context to <server> (namespace calamari-testing) using EKS cluster name my-eks-cluster
 [Verbose] Attempting to authenticate with aws-cli
-[Verbose] Unable to authenticate to <server> using the aws cli. Failed with error message: Error reading JObject from JsonReader. Path '', line 0, position 0.
+[Verbose] Unable to authenticate to <server> using the aws cli. Failed with error message: Unexpected character encountered while parsing value: P. Path '', line 0, position 0.
```
This test apparently was always failing to parse the authentication object because it passes in a bogus region `<server>` so I just updated the error as it's changed now that I have joined all the log lines before parsing.

johnsimons commented:
@scottmerchant I think the issue is that we are not using a fixed version for `awscli`.
This version [here](https://github.com/OctopusDeploy/Calamari/blob/303bf359f5c5135aab080fc4b48f57927b7c1c01/source/Calamari.Tests/KubernetesFixtures/InstallTools.cs#L136) is not being used as part of the URL.
According to the [doco](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-version.html), the URL should be something like https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.0.30.zip

scottmerchant commented:
> @scottmerchant I think the issue is that we are not using a fixed version for `awscli`. This version [here](https://github.com/OctopusDeploy/Calamari/blob/303bf359f5c5135aab080fc4b48f57927b7c1c01/source/Calamari.Tests/KubernetesFixtures/InstallTools.cs#L136) is not being used as part of the URL. According to the [doco](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-version.html), the URL should be something like https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.0.30.zip

Ah yep good idea. I'll add a change to make it fixed in this PR.

scottmerchant commented:
@johnsimons I've updated the install tools area to download a specific version of the CLI and I've updated it to download the latest version `v2.11.0`.

scottmerchant commented on source/Calamari.Tests/KubernetesFixtures/InstallTools.cs:
```diff
@@ -133,8 +133,8 @@ public async Task InstallAwsCli()
-                        var version = "2.8.3";
-                        return Task.FromResult((version, GetAwsCliDownloadLink()));
+                        var version = "latest";
+                        return Task.FromResult((version, GetAwsCliDownloadLink(version)));
```
Good point from Yihao was that we want this test to fail if latest aws cli doesn't work with our stuff so I updated this to download the latest but you can optionally change it on your dev machine if you want to test a specific version.

Note: on mac it doesn't install the version it downloads - you need to manually install the version you want to run. 🤷‍♂️

johnsimons replied:
it feels like we have a mix of testing low and high ranges

scottmerchant replied:
I suppose we could fix the version as long as it's noted in the doco.

johnsimons replied:
Aren't the other versions fixed for the other tooling?

johnsimons replied:
Maybe something to ask others, what is our stance on this?

scottmerchant replied:
I have fixed the version to what it was meant to be fixed to originally. Can you please give it a final check over and let me know if there are any other issues? 🙏
